app/src/main/java/cm/aptoide/pt/preferences/AptoideMd5Manager.java

```diff
@@ -41,18 +41,16 @@ public String getAptoideMd5() {
     if (cachedMd5 != null) return cachedMd5;
 
     String savedMd5 = parseMd5(preferencesPersister.get(APTOIDE_MD5, ""), vercode);
-    if (!savedMd5.isEmpty()) {
-      cachedMd5 = savedMd5;
-    } else {
-      cachedMd5 = calculateMd5Sum();
-      persistAptoideMd5(vercode, cachedMd5);
-    }
-    return cachedMd5;
+
+    return savedMd5;
   }
 
-  private String calculateMd5Sum() {
+  public Void calculateMd5Sum() {
+    if (cachedMd5 != null) return null;
+    if (!parseMd5(preferencesPersister.get(APTOIDE_MD5, ""), vercode).isEmpty()) return null;
```

Comment:
It would be nice if we cache this value to memory if there is md5 in shared preferences. Right now we have to always fetch from shared preferences if the md5 isn't calculated...

Reply:
fixed